OpenRA.Mods.Common/Lint/CheckTranslationReference.cs

```diff
@@ -25,10 +29,115 @@ sealed class CheckTranslationReference : ILintPass, ILintMapPass
 	{
 		const BindingFlags Binding = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static;
 
+		// These widget IDs are known to be called via Game.OpenWindow or similar, and thus they use the available window space.
+		// HACK: We hardcode this list, rather than attempting to work it out dynamically.
+		static readonly string[] KnownWidowWidgetIds = new[]
+		{
+			// HACK: LABEL_DROPDOWN_TEMPLATE can be parented under many different dropdown buttons,
+			// instead of building the proper list in KnownParentChildWidgetIds,
+			// just assume the window size instead.
+			"LABEL_DROPDOWN_TEMPLATE",
+
+			"MAINMENU",
+			"INGAME_ROOT",
+			"EDITOR_ROOT",
+			"GAMESAVE_LOADING_SCREEN",
+			"FMVPLAYER",
+			"MAINMENU_INTRODUCTION_PROMPT",
+			"MAINMENU_SYSTEM_INFO_PROMPT",
+			"SETTINGS_PANEL",
+			"CREDITS_PANEL",
+			"SERVER_LOBBY",
+			"MAPCHOOSER_PANEL",
+			"MULTIPLAYER_PANEL",
+			"MULTIPLAYER_CREATESERVER_PANEL",
+			"DIRECTCONNECT_PANEL",
+			"CONNECTIONFAILED_PANEL",
+			"CONNECTING_PANEL",
+			"CONNECTION_SWITCHMOD_PANEL",
+			"REPLAYBROWSER_PANEL",
+			"GAMESAVE_BROWSER_PANEL",
+			"MUSIC_PANEL",
+			"ASSETBROWSER_PANEL",
+			"MISSIONBROWSER_PANEL",
+			"FULLSCREEN_PLAYER",
+			"THREEBUTTON_PROMPT",
+			"TWOBUTTON_PROMPT",
+			"TEXT_INPUT_PROMPT",
+			"NEW_MAP_BG",
+			"SAVE_MAP_PANEL",
+			"ENCYCLOPEDIA_PANEL",
+		};
+
+		// Various logic classes dynamically attach children to parents at runtime.
+		// HACK: We hardcode a list of the attachments made via code, so we can verify the sizes of the child within their parent container.
+		static readonly Dictionary<(string Logic, string ParentWidgetId), string[]> KnownParentChildWidgetIds = new()
+		{
+			{ ("LoadIngameChatLogic", "CHAT_ROOT"), new[] { "CHAT_PANEL" } },
```

Comment:
I'm wondering if it would be cleaner to pull these out into a yaml file?

Reply:
An easier and IMO better approach would be to move it onto the chrome logic, like we do with hotkey names.

Reply:
Yes having an attribute that allows logic classes to specify the dynamic parent->child relationships they are going to set up would be the scalable solution here. Some of those are buried deep in helper methods though, so I don't think it'd be a quick refactoring for all cases.

Would we be okay to defer that as a follow-up, or should we get that in as a pre-req for this PR?

Reply:
I'm fine with it being a followup.